contracts/src/core/BatcherPayments.sol

```diff
@@ -0,0 +1,145 @@
+    event CreatedNewTask(
+        bytes32 batchMerkleRoot,
+        string batchDataPointer,
+        address[] proofSubmitters,
+        uint256 costOfRespondToTask
+    );
+
+    // STORAGE
+    address public AlignedLayerServiceManager;
+    address public BatcherWallet;
+
+    mapping(address => uint256) public UserBalances;
+
+    uint256 public this_tx_base_gas_cost;
+    uint256 public create_task_gas_price;
+    uint256 public extra_user_tx_gas_cost;
+
+    // storage gap for upgradeability
+    uint256[25] private __GAP;
+
+    // CONSTRUCTOR & INITIALIZER
+    constructor() {
+        _disableInitializers();
+    }
+    
+    function initialize (
+        address _AlignedLayerServiceManager,
+        address _BatcherWallet, 
+        uint256 _this_tx_base_gas_cost, 
+        uint256 _create_task_gas_price,
+        uint256 _extra_user_tx_gas_cost
+    ) public initializer {
+        __Ownable_init(); // default is msg.sender
+
+        AlignedLayerServiceManager = _AlignedLayerServiceManager;
+        BatcherWallet = _BatcherWallet;
+        this_tx_base_gas_cost = _this_tx_base_gas_cost;
+        create_task_gas_price = _create_task_gas_price;
+        extra_user_tx_gas_cost = _extra_user_tx_gas_cost;
+    }
+
+    // PAYABLE FUNCTIONS
+    receive() external payable {
+        UserBalances[msg.sender] += msg.value;
+        emit PaymentReceived(msg.sender, msg.value);
+    }
+
+    // PUBLIC FUNCTIONS
+    function createNewTask(
+        bytes32 batchMerkleRoot,
+        string calldata batchDataPointer,
+        address[] calldata proofSubmitters, // one address for each payer proof, 1 user has 2 proofs? send twice that address
+        uint256 costOfRespondToTask // TODO hardcode gas cost? It is variable because of signature sdk. could have upper bound and multiply by current gas cost + x%. 
+    ) external onlyBatcher whenNotPaused {
+        uint256 amountOfSubmitters = proofSubmitters.length;
+        require(amountOfSubmitters > 0, "No proof submitters");
+        
+        // uint256 this_tx_base_gas_cost = 42000; // base gas cost of this transaction, without createNewTask or users to iterate
+        // uint256 create_task_gas_price = 60000; // gas price of createNewTask in in AlignedLayerServiceManager
+        // // uint256 respond_task_gas_price = 250000; // gas price of respondToTask in in AlignedLayerServiceManager
+        // uint16 extra_user_tx_gas_cost = 6500; // upper bound of gas cost of adding a user
+
+        // each user must pay its fraction of the gas cost of this transaction back to the batcher
+        // + 10% for increments in gas price
+        uint256 cost_of_this_tx = ((this_tx_base_gas_cost + create_task_gas_price + (extra_user_tx_gas_cost * amountOfSubmitters)) * tx.gasprice * 11) / 10;
+
+        // divide the price by the amount of submitters
+        uint256 totalCostPerProof = (costOfRespondToTask + cost_of_this_tx) / amountOfSubmitters;
```

Comment:
mix of camel and snake case, should probably use snakeCase for everything

Reply:
How about now?
https://docs.soliditylang.org/en/latest/style-guide.html#constants